### PR TITLE
Fix problem with steps counter

### PR DESCRIPTION
**Pull request by evgeniy1801**
[Issue #107 ](https://github.com/nathanboktae/mocha-casperjs/issues/107)

### CODE REVIEW

File: mocha-casperjs.js [hunk 1]
```diff
@@ -48,7 +48,6 @@ module.exports = function (Mocha, casper, utils) {
   })
 
   casper.on('waitFor.timeout', function(timeout, details) {
-    casper.step++
```
Comment (goganchic) [inline mocha-casperjs.js]: there are two more places with `casper.step++`. Are these commands required? Should it be deleted too?

Comment (evgeniy1801) [inline mocha-casperjs.js]: I deleted it in the `step.timeout` handler. If I delete it in the `timeout` handler, some tests will fail.

Comment (nathanboktae) [inline mocha-casperjs.js]: Yeah `timeout` is like a page timeout, not of the `waitFor` variety. It's a bit different.

File: test/all.coffee [hunk 2]
```diff
@@ -496,4 +496,15 @@ describe 'mocha-casperjs', ->
           results.stats.failures.should.equal 0
           done()
 
+  describe 'Steps checking', ->
+    it 'should run all steps if previous test failed', (done) ->
+      runMochaCasperJsTest
+        before: ->
+          casper.start()
+        test: 'all-steps-running.js'
+      , (output) ->
```
Comment (nathanboktae) [inline test/all.coffee]: Also assert that `code` is 1. Thanks!

File: mocha-casperjs.js [hunk 3]
```diff
@@ -47,8 +47,13 @@ module.exports = function (Mocha, casper, utils) {
     })
   })
 
+  function clearSteps() {
+    casper.step = 0
+    casper.steps = []
+  }
+
   casper.on('waitFor.timeout', function(timeout, details) {
-    casper.step++
```
Comment (evgeniy1801) [inline mocha-casperjs.js]: Test should be aborted if it fails. This line should do the trick, but  it is not enough. If test fails on some sub step (level > 1), current test will skip current step and will continue after it.

**Comment by evgeniy1801** [comment]
@nathanboktae, it's all right? Would you like some changes?

File: mocha-casperjs.js [hunk 4]
```diff
@@ -47,8 +47,13 @@ module.exports = function (Mocha, casper, utils) {
     })
   })
 
+  function clearSteps() {
+    casper.step = 0
+    casper.steps = []
+  }
```
Comment (nathanboktae) [inline mocha-casperjs.js]: no no you are completely changing the functionality here, and deeply mucking with internal state of casper by clearing all the steps. `failTests` already handles when the steps are behind.

Comment (goganchic) [inline mocha-casperjs.js]: Can we just create new instance of casper for every test? In this case we will not change something inside of casper and from the other side will fix problem with steps skipping.

Comment (nathanboktae) [inline mocha-casperjs.js]: No we can't.

Comment (goganchic) [inline mocha-casperjs.js]: Would you please explain me why?

**Comment by evgeniy1801** [comment]
I made 2 commits: the first one fixes the problem with skipping the first step in the test after the failure. In the second - the solution of the problem with the continuation of the test after its failure.

**Comment by nathanboktae** [comment]
OK thanks I like this version a lot better since it's using a public API of `casper`. Thanks also for the extended tests!

It's a change of behavior in how steps are skipped on failed so I'll have to go to `0.6.0`. Let me review it more in the next day or two to find anything else, thanks.

**Comment by nathanboktae** [comment]
Thanks! I compressed the 6 tests down to 4 and remove the google dependency (I don't even think they support http anymore, thanks 'murica). If CI is good I'll release `0.6.0` soon.

**Pull request merged**